VoodooPS2Trackpad/VoodooPS2ALPSGlidePoint.cpp

```diff
@@ -45,220 +177,231 @@ UInt32 ApplePS2ALPSGlidePoint::deviceType()
 UInt32 ApplePS2ALPSGlidePoint::interfaceID()
 { return NX_EVS_DEVICE_INTERFACE_BUS_ACE; };
 
-IOItemCount ApplePS2ALPSGlidePoint::buttonCount() { return 2; };
-IOFixed     ApplePS2ALPSGlidePoint::resolution()  { return _resolution; };
-bool IsItALPS(ALPSStatus_t *E6,ALPSStatus_t *E7);
+IOItemCount ApplePS2ALPSGlidePoint::buttonCount() { return _buttonCount; };
+IOFixed     ApplePS2ALPSGlidePoint::resolution()  { return _resolution << 16; };
+
+#define abs(x) ((x) < 0 ? -(x) : (x))
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
-bool ApplePS2ALPSGlidePoint::init(OSDictionary * dict)
-{
+bool ApplePS2ALPSGlidePoint::init(OSDictionary *dict) {
+    
     //
     // Initialize this object's minimal state. This is invoked right after this
     // object is instantiated.
     //
     
-    if (!super::init(dict))
+    if (!super::init(dict)) {
         return false;
-
+    }
+    
     // initialize state...
-    _device                    = 0;
-    _interruptHandlerInstalled = false;
-    _packetByteCount           = 0;
-    _resolution                = (100) << 16; // (100 dpi, 4 counts/mm)
-    _touchPadModeByte          = kTapEnabled;
-    _scrolling                 = SCROLL_NONE;
-    _zscrollpos                = 0;
+    for (int i = 0; i < MAX_TOUCHES; i++)
+        fingerStates[i].virtualFingerIndex = -1;
+    
+    memset(freeFingerTypes, true, kMT2FingerTypeCount);
+    freeFingerTypes[kMT2FingerTypeUndefined] = false;
+    
+    // announce version
+    extern kmod_info_t kmod_info;
+    DEBUG_LOG("ALPS: Version %s starting on OS X Darwin %d.%d.\n", kmod_info.version, version_major, version_minor);
+    
+    setProperty("Revision", 24, 32);
     
     return true;
 }
 
-// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void ApplePS2ALPSGlidePoint::injectVersionDependentProperties(OSDictionary *config) {
+    // inject properties specific to the version of Darwin that is runnning...
```

Comment:
Probably not, but since Elan and Synaptics have it to inject some properties, I added it as well.

![image](https://user-images.githubusercontent.com/28839925/153755129-e39ae469-7672-4f10-80dc-22ae91e67fe2.png)